File: api/Model/Transaction.cs
namespace api.Model;

public class Transaction
{
    public string Id { get; set; }
    public string Title { get; set; }
    public string? Description { get; set; }
    public int Amount { get; set; }
    public DateTime Created { get; set; } = DateTime.Now;
    public TransactionType TransactionType { get; set; }
    public TransactionCategory Category { get; set; }
    
    public User User { get; set; }
    public Account Account { get; set; }
}

public enum TransactionType
{
    Income,
    Expense,
    Transfer,
}

public enum TransactionCategory
{
    Food,
    Shopping,
    Bills,
    Groceries,
    Home,
    Health,
    Other,
}

File: src/types/transaction.ts
export interface Transaction {
  id: string;
  title: string | null;
  description: string | null;
  amount: number;
  type: "Income" | "Expense" | "Transfer";
  meta: "default" | "recurring" | "goal" | "subscription" | "loan";
  picture: Uint8Array | null;
  created: string | null;
  accountId: string | null;
  categoryId: number | null;
  goalId?: string | null;
  subscriptionId?: string | null;
  loanId?: string | null;
  category?: Category;
  account?: Account;
}

export interface TransactionFormData {
  title: string;
  description: string;
  amount: number;
  type: "Income" | "Expense";
  categoryId: number;
  accountId: string;
  meta: "default" | "recurring" | "goal" | "subscription" | "loan";
}

export interface GroupedTransaction {
  date: string;
  transactions: Transaction[];
  dayTotal: number;
}

// Import types that Transaction depends on
import type { Category } from "./category";
import type { Account } from "./account";